content/docs/user-guide/programming/autogen/_index.md

```diff
@@ -1,10 +1,255 @@
+
+  1. For each XML or JSON data file:
+
+        1. The data is processed through a single Jinja template. Inside of the template, Jinja has access to a native Python object that represents the contents of the input file.
+
+        1. The output is written to its corresponding file.
+
+
+## Authoring Jinja templates and data inputs
+
+How you use AzAutoGen in your Gem or project is driven by the functionality that you author in the Jinja template and the data inputs that you supply. 
+
+Before you use the Jinja template, it helps to understand some Jinja concepts. 
+A *Jinja template* is a simple text file that takes in data, replaces parts of the template with the provided data, and outputs the final file. 
+A template contains *variables* and *expressions*, which the data replaces, as well as *tags*, which are logical instructions on how the output file should appear. *Data* are values that are input into the template. For AzAutoGen, you can author data in XML (`.xml`) or JSON (`.json`) files. 
+
+Jinja2's templating system can also work with a set of data files to generate a set of output files, making AzAutoGen an efficient solution if you need to generate many files.
+
+For more information about authoring Jinja templates and data input, refer to the [Template Designer Documentation](https://jinja.palletsprojects.com/templates/) on the Jinja website.
+
+### Input mapping and variables
+
+As part of the Jinja2 template engine, AzAutoGen can access the data that's exposed to the templating system as Python objects. JSON input files are loaded directly as Python dictionaries, and XML files are represented by Python's [xml.etree.ElementTree](https://docs.python.org/3/library/xml.etree.elementtree.html) objects.
+
+When authoring a Jinja template, you can use the following variables. AzAutoGen defines these variables and its attributes, sets their values, and then sends it through Jinja's templating system to generate the output files.
+
+| Name | Value |
+|-|-|
+| `dataFiles` | An array of dictionaries containing the objects that are read from input files. |
+| `dataFileNames` | An array of the input file names. Neither `dataFiles` nor `dataFileNames` are guaranteed to be ordered, but `dataFileNames[n]` is always the source for the object available in `dataFiles[n]`. |
+| `templateName` | The name of the template file that AzAutoGen is currently processing. |
+| `outputFile` | The name of the output file. |
+| `filename` | The name of the file that AzAutoGen is currently generating. |
+
+### XML validation
+
+If you author input data files in XML, you can bind your `.xml` file to an `.xsd` file. Then, AzAutoGen will detect the `.xsd` file and validate your data input files. If it doesn't conform, AzAutoGen initiates a build failure.
```

Comment:
I don't believe any of this is actually hooked up at the moment. It used to be a feature of the autogen system, but when we moved to the new python environment provided as part of the cmake port, we lost access to lxml.

Reply:
Removing this section